### PR TITLE
Add IsFinite operator

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
- [x] IsFinite operator specification
- [x] IsFinite core operator (without the evaluate method)
- [x] Python API for the IsFinite
- [x] ONNX support for IsFinite
- [x] required tests

### Tickets:
 - 93376

### CODE REVIEW

File: src/core/tests/type_prop/is_finite.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+TEST(type_prop, isfinite_output_shape) {
+    const auto data = make_shared<Parameter>(ov::element::f16, ov::Shape{4, 2});
+    const auto isfinite = make_shared<IsFinite>(data);
+
+    EXPECT_EQ(isfinite->get_output_partial_shape(0), ov::PartialShape({4, 2}))
+        << "The output shape of IsFinite is incorrect";
+    ASSERT_EQ(isfinite->get_shape(), (ov::Shape{4, 2})) << "The output shape of IsFinite is incorrect";
+}
+
+TEST(type_prop, isfnite_sample_dynamic_batch) {
```
Comment (mitruska) [inline src/core/tests/type_prop/is_finite.cpp:21]: Interval dimensions should be tested too, please take a look at the (https://github.com/openvinotoolkit/openvino/pull/13369#discussion_r991533375)

File: src/core/tests/type_prop/is_finite.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    const auto data = make_shared<Parameter>(ov::element::f16, ov::PartialShape{ov::Dimension::dynamic(), 21, 37});
+    const auto isfinite = make_shared<IsFinite>(data);
+
+    EXPECT_EQ(isfinite->get_output_partial_shape(0), ov::PartialShape({ov::Dimension::dynamic(), 21, 37}))
+        << "The output shape of IsFinite is incorrect";
+}
+
+TEST(type_prop, isfinite_output_type) {
```
Comment (mitruska) [inline src/core/tests/type_prop/is_finite.cpp:29]: Test for incompatible input type should be added too.
In general these `type_prop` tests could be shared between `IsNaN` and `IsInf` op.

File: src/core/src/op/is_finite.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    OV_OP_SCOPE(v10_IsFinite_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return std::make_shared<IsFinite>(new_args.at(0));
+}
+
+void ov::op::v10::IsFinite::validate_and_infer_types() {
+    OV_OP_SCOPE(v10_IsFinite_validate_and_infer_types);
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0).is_real(),
+                          "The element type of the input tensor must be a floating point number.");
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply src/core/src/op/is_finite.cpp:24]: Yeah, I guess do. I will update that.

File: src/core/tests/CMakeLists.txt [hunk 4]
```diff
@@ -169,6 +169,7 @@ set(SRC
     type_prop/if.cpp
     type_prop/interpolate.cpp
     type_prop/irdft.cpp
+    type_prop/is_finite.cpp
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply src/core/tests/CMakeLists.txt:172]: done

File: src/core/include/openvino/op/is_finite.hpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#pragma once
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v10 {
+/// \brief Boolean mask that maps NaN and Infinity values to false and other values to true.
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API IsFinite : public Op {
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply src/core/include/openvino/op/is_finite.hpp:14]: done

File: src/core/include/openvino/opsets/opset10_tbl.hpp [hunk 6]
```diff
@@ -48,6 +48,7 @@ _OPENVINO_OP_REG(GroupConvolution, ov::op::v1)
 _OPENVINO_OP_REG(GroupConvolutionBackpropData, ov::op::v1)
 _OPENVINO_OP_REG(GRN, ov::op::v0)
 _OPENVINO_OP_REG(HardSigmoid, ov::op::v0)
+_OPENVINO_OP_REG(IsFinite, ov::op::v10)
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply src/core/include/openvino/opsets/opset10_tbl.hpp]: done

File: src/core/tests/visitors/op/is_finite.cpp [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "unary_ops.hpp"
+#include "ngraph/opsets/opset10.hpp"
+
+using Types = ::testing::Types<UnaryOperatorType<ngraph::op::v10::IsFinite, ngraph::element::f32>>;
+                            //    UnaryOperatorType<ngraph::op::v10::IsFinite, ngraph::element::i32>>;
```
Comment (mitruska) [inline src/core/tests/visitors/op/is_finite.cpp:9]: This commented line should be removed
```suggestion
```

File: src/core/tests/visitors/op/is_finite.cpp [hunk 8]
```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "unary_ops.hpp"
+#include "ngraph/opsets/opset10.hpp"
```
Comment (mitruska) [inline src/core/tests/visitors/op/is_finite.cpp:6]:
```suggestion
#include "openvino/opsets/opset10.hpp"
```

File: src/core/tests/visitors/op/is_finite.cpp [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+using Types = ::testing::Types<UnaryOperatorType<ngraph::op::v10::IsFinite, ngraph::element::f32>>;
```
Comment (mitruska) [inline src/core/tests/visitors/op/is_finite.cpp:8]: It's better to use `ov` namespace instead `ngraph` in the new tests

File: docs/ops/comparison/IsFinite_10.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+**Category**: *Comparison*
+
+**Short description**: *IsFinite* performs element-wise test for finiteness (not Infinity and not NaN) on elements of a given input tensor. It returns the boolean mask of a given tensor which maps
+`NaN` and `Infinity` to `False` and all other values to `True`
+*IsFinite* operation has no attributes.
+
+**Detailed description**: *IsFinite* returns the bolean mask of the input tensor in which `False` corresponds to `NaN` and `Infinity` and `True` to all other values.
+* The output tensor has the same shape as input tensor.
+* The `i`'th element of the output tensor is `False` if `i`'th element of the input tensor is `NaN` or `Inf`. Otherwise it is `True`.
+* For example, for given input tensor `[NaN, 2.1, 3.7, Inf]` the output tensor is `[False, True, True, False]`.
```
Comment (mitruska) [inline docs/ops/comparison/IsFinite_10.md:14]: (Optional) Example values can be also added to the "Examples" section.
Reference: [Select_1.md](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/condition/Select_1.md), [CumSum_3.md](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/arithmetic/CumSum_3.md)

File: src/core/tests/type_prop/is_finite.cpp (hunk 2 above)
Comment (mitruska) [reply src/core/tests/type_prop/is_finite.cpp:28]: `IsNaN` type_prop tests as example: https://github.com/openvinotoolkit/openvino/blob/d146f945cbb1c78e8046d7b1330ee7b61fde18aa/src/core/tests/type_prop/is_nan.cpp#L52-L60

File: docs/ops/comparison/IsFinite_10.md (hunk 10 above)
Comment (bszmelcz) [reply docs/ops/comparison/IsFinite_10.md:14]: done

File: src/core/tests/type_prop/is_finite.cpp [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+TEST(type_prop, isfinite_sample_dynamic_shape) {
+    const auto data = std::make_shared<Parameter>(ov::element::f16, ov::PartialShape::dynamic(5));
+    const auto isfinite = std::make_shared<IsFinite>(data);
+
+    EXPECT_EQ(isfinite->get_output_partial_shape(0), (ov::PartialShape::dynamic(5)))
+        << "The output shape of IsFinite is incorrect";
+}
+
+TEST(type_prop, isfinite_bad_input_type) {
```
Comment (mitruska) [inline src/core/tests/type_prop/is_finite.cpp:52]: (Optional) `bad` -> `incorrect` ? ;)
```suggestion
TEST(type_prop, isfinite_incorrect_input_type) {
```

**Review by mitruska — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

File: src/frontends/onnx/frontend/src/ops_bridge.cpp [hunk 12]
```diff
@@ -358,6 +359,7 @@ OperatorsBridge::OperatorsBridge() {
     REGISTER_OPERATOR("If", 1, if_op);
     REGISTER_OPERATOR("ImageScaler", 1, image_scaler);
     REGISTER_OPERATOR("InstanceNormalization", 1, instance_norm);
+    REGISTER_OPERATOR("IsFinite", 1, is_finite)
```
Comment (mmikolajcz) [inline src/frontends/onnx/frontend/src/ops_bridge.cpp:362]: Based on ONNX documentation, IsFinite is not an ONNX operator. Maybe it shouldn't be registered?

File: src/frontends/onnx/tests/models/is_finite.prototxt [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    type {
+      tensor_type {
+        elem_type: 9
+        shape {
+          dim {
+            dim_value: 1
+          }
+          dim {
+            dim_value: 3
+          }
```
Comment (mmikolajcz) [inline src/frontends/onnx/tests/models/is_finite.prototxt:37]: Dim mismatch between prototxt and onnx_import test
```suggestion
        shape {
          dim {
            dim_value: 1
          }
          dim {
            dim_value: 2
          }
          dim {
            dim_value: 3
          }
        }
      }
    }
  }
  output {
    name: "y"
    type {
      tensor_type {
        elem_type: 9
        shape {
          dim {
            dim_value: 1
          }
          dim {
            dim_value: 2
          }
          dim {
            dim_value: 3
          }
```

File: src/frontends/onnx/frontend/src/ops_bridge.cpp (hunk 12 above)
Comment (bszmelcz) [reply src/frontends/onnx/frontend/src/ops_bridge.cpp]: done

**Review by rkazants — CHANGES_REQUESTED** [review]

File: src/core/src/op/is_finite.cpp [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+std::shared_ptr<ov::Node> ov::op::v10::IsFinite::clone_with_new_inputs(const OutputVector& new_args) const {
+    OV_OP_SCOPE(v10_IsFinite_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return std::make_shared<IsFinite>(new_args.at(0));
+}
+
+void ov::op::v10::IsFinite::validate_and_infer_types() {
+    OV_OP_SCOPE(v10_IsFinite_validate_and_infer_types);
+    element::Type input_element_type = get_input_element_type(0);
+    element::Type output_element_type = ov::element::Type_t::boolean;
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply src/core/src/op/is_finite.cpp]: done

**Review by rkazants — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

**Pull request merged**